#8 set up qulice and test coverage

olenagerasimova opened the pull request:
For #8 I've set up qulice and test coverage, extended test for `GhThread` and added todo to continue increasing code coverage.

0crat commented:
Job [#27](https://github.com/g4s8/ghman/issues/27) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/CJT4R32Q3/858fbc72-b7d9-4a1d-aa84-adcdb2f85f7c, version: 0.47.5, hash: ${buildNumber} -->

0crat commented:
This pull request [#27](https://github.com/g4s8/ghman/issues/27) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CJT4R32Q3/5d35527b-3e96-41f2-af41-4233d24d93bb); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

<!-- https://www.0crat.com/footprint/CJT4R32Q3/0486695e-cc63-4d6f-8567-68379cd21654, version: 0.47.5, hash: ${buildNumber} -->

victornoel commented on src/test/java/com/g4s8/ghman/user/GhThreadTest.java:
```diff
@@ -32,11 +34,48 @@
         final String tid = "4";
-        MatcherAssert.assertThat(
+        new Assertion<>(
+            "Failed to get thread id from json",
```
@olenagerasimova the preferred wording for assertions is the positive one, so here it should be "must get thread id from json".Could you update this one and the other tests here?

victornoel commented on pom.xml:
```diff
@@ -25,6 +25,9 @@
     <telegram.version>3.6</telegram.version>
+    <surefire.version>3.0.0-M1</surefire.version>
+    <junit.version>5.4.2</junit.version>
+    <cactoos.dir>${basedir}/src/main/java/org/cactoos</cactoos.dir>
```
@olenagerasimova this is used nowhere

victornoel commented on pom.xml:
```diff
@@ -25,6 +25,9 @@
   <version>1.0-SNAPSHOT</version>
   <properties>
     <telegram.version>3.6</telegram.version>
+    <surefire.version>3.0.0-M1</surefire.version>
```
@olenagerasimova it was 3.0.0-M3, why switch back to M1?

victornoel commented on pom.xml:
```diff
@@ -151,11 +149,6 @@
     </dependency>
-    <dependency>
-      <groupId>org.glassfish</groupId>
-      <artifactId>javax.json</artifactId>
```
@olenagerasimova why was this removed?

olenagerasimova replied:
@victornoel It was a duplicate

victornoel commented on pom.xml:
```diff
@@ -261,9 +263,13 @@
             <configuration>
               <output>file</output>
+              <excludes>
+                <exclude>**/*.sketch</exclude>
```
@olenagerasimova why exclude this? there are no `.sketch` file in this repository, is there?

victornoel commented on pom.xml:
```diff
@@ -261,9 +263,13 @@
+              <excludes>
+                <exclude>**/*.sketch</exclude>
+                <exclude>**/org/cactoos/**/*</exclude>
+                <exclude>**/ghman/data/*</exclude>
```
@olenagerasimova why is this excluded?

olenagerasimova replied:
@victornoel it's database related package, I do not think there will be any unit tests for classes of this package. But actually it might be unfair to exclude it from coverage metrics. @g4s8 what do say? Should we exclude it or not?

victornoel replied:
@olenagerasimova I don't see why there would be a different treatment for this particular package. Let's wait for @g4s8 ruling on this anyway :)

g4s8 replied:
@victornoel @olenagerasimova I think we can test `data` package with integration tests and then add it back to coverage metrics, WDYT?

victornoel replied:
@g4s8 @olenagerasimova I just don't understand why we would need to first exclude them.

olenagerasimova replied:
@victornoel @g4s8 I just remove this line to save our time

victornoel commented on pom.xml:
```diff
@@ -272,54 +278,33 @@
                       <element>BUNDLE</element>
+                      <!--
+                        @todo #8:30min Increase code coverage by
+                         writing unit tests. All coverage ration minimums are
```
@olenagerasimova ration -> ratio

olenagerasimova commented:
@victornoel I've made correction and added some comments. Also I've removed puzzle for this ticket (forget to do it earlier) and unnecessary `JavadocMethodCheck` suppression, qulice now understands it's a test and does not complain.  

As for this 
> Will this be visible directly from the PR with codecov or something like that or is this something else to setup on top of this?

I am afraid I do not understand what are toking about.

victornoel commented:
@olenagerasimova thanks

@g4s8 apart from https://github.com/g4s8/ghman/pull/27#discussion_r292584193, it should be alright.
Also maybe it is needed to setup something specific to get feedback about coverage in PRs? cf discussion above.

olenagerasimova commented:
@victornoel I've removed `data` package exclusion

olenagerasimova commented:
@victornoel ping

victornoel commented:
@g4s8 it's alright with me

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/19219-502378721)

rultor commented:
> @rultor merge

@g4s8 @olenagerasimova Oops, I failed. You can see the full log [here](http://www.rultor.com/t/19219-502378721) (spent 2min)

```
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-booter/3.0.0-M3/surefire-booter-3.0.0-M3.jar (312 kB at 2.0 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/ow2/asm/asm/7.0-beta/asm-7.0-beta.jar (114 kB at 638 kB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/maven-compat/3.0/maven-compat-3.0.jar (285 kB at 1.6 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/maven-surefire-common/3.0.0-M3/maven-surefire-common-3.0.0-M3.jar (627 kB at 2.5 MB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit-platform/3.0.0-M3/surefire-junit-platform-3.0.0-M3.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit-platform/3.0.0-M3/surefire-junit-platform-3.0.0-M3.jar (20 kB at 923 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit-platform/3.0.0-M3/surefire-junit-platform-3.0.0-M3.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-junit-platform/3.0.0-M3/surefire-junit-platform-3.0.0-M3.pom (5.5 kB at 275 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-providers/3.0.0-M3/surefire-providers-3.0.0-M3.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/surefire-providers/3.0.0-M3/surefire-providers-3.0.0-M3.pom (2.5 kB at 121 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/common-java5/3.0.0-M3/common-java5-3.0.0-M3.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/common-java5/3.0.0-M3/common-java5-3.0.0-M3.pom (3.1 kB at 146 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-launcher/1.3.1/junit-platform-launcher-1.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-launcher/1.3.1/junit-platform-launcher-1.3.1.pom (2.2 kB at 89 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-engine/1.3.1/junit-platform-engine-1.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-engine/1.3.1/junit-platform-engine-1.3.1.pom (2.4 kB at 126 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-commons/1.3.1/junit-platform-commons-1.3.1.pom
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-commons/1.3.1/junit-platform-commons-1.3.1.pom (2.0 kB at 106 kB/s)
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/common-java5/3.0.0-M3/common-java5-3.0.0-M3.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-launcher/1.3.1/junit-platform-launcher-1.3.1.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-engine/1.3.1/junit-platform-engine-1.3.1.jar
[INFO] Downloading from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-commons/1.3.1/junit-platform-commons-1.3.1.jar
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/apache/maven/surefire/common-java5/3.0.0-M3/common-java5-3.0.0-M3.jar (32 kB at 1.4 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-launcher/1.3.1/junit-platform-launcher-1.3.1.jar (95 kB at 1.9 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-commons/1.3.1/junit-platform-commons-1.3.1.jar (78 kB at 1.7 MB/s)
[INFO] Downloaded from central: https://repo.maven.apache.org/maven2/org/junit/platform/junit-platform-engine/1.3.1/junit-platform-engine-1.3.1.jar (135 kB at 2.0 MB/s)
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.g4s8.ghman.env.EnvironmentVariablesTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.073 s - in com.g4s8.ghman.env.EnvironmentVariablesTest
[INFO] Running com.g4s8.ghman.user.GhThreadTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.055 s - in com.g4s8.ghman.user.GhThreadTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:check (jacoco-check) @ ghman ---
[INFO] Loading execution data file /home/r/repo/target/jacoco.exec
[INFO] Analyzed bundle 'ghman' with 21 classes
[WARNING] Rule violated for bundle ghman: classes missed count is 19, but expected maximum is 18
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 35.221 s
[INFO] Finished at: 2019-06-15T16:13:46Z
[INFO] Final Memory: 31M/378M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.jacoco:jacoco-maven-plugin:0.8.4:check (jacoco-check) on project ghman: Coverage checks have not been met. See log for details. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
container 0089581f67affa4eea28b7f7fc92dce0f6c231b9c0bdebff392e2c7d70f84c29 is dead
Sat Jun 15 18:14:32 CEST 2019
```

0crat commented:
@victornoel[/z](https://www.0crat.com/u/victornoel) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

<!-- https://www.0crat.com/footprint/CJT4R32Q3/9a9e9e7d-e8bf-45df-a562-d35f5c76142b, version: 0.47.6, hash: ${buildNumber} -->

olenagerasimova commented:
@g4s8 there are problems with puzzles from #2. What do we do here?

g4s8 commented:
@olenagerasimova we're waiting for @samihibrahim to fix puzzles in #2

olenagerasimova commented:
@g4s8 please, try to merge, I've pulled master

g4s8 commented:
@rultor merge

rultor commented:
> @rultor merge

@g4s8 OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/19219-502692055)

rultor commented:
> @rultor merge

@g4s8 Done! FYI, the full log is [here](http://www.rultor.com/t/19219-502692055) (took me 3min)

0crat commented:
@ammaratef45[/z](https://www.0crat.com/u/ammaratef45) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/CJT4R32Q3/7e7d1abf-7c5f-4bf0-9f2f-3abbdd8fa8c0, version: 0.47.6, hash: ${buildNumber} -->

0crat commented:
Job [#27](https://github.com/g4s8/ghman/issues/27) is not in the agenda of @victornoel[/z](https://www.0crat.com/u/victornoel), can't inspect

<!-- https://www.0crat.com/footprint/CJT4R32Q3/f170e1f5-c40a-49d0-9a86-453698326885, version: 0.47.6, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

<!-- https://www.0crat.com/footprint/CJT4R32Q3/656a75ef-d209-448b-9d72-fb856bd6c3a2, version: 0.47.6, hash: ${buildNumber} -->

ammaratef45 commented:
@0crat quality good

0crat commented:
Quality review completed: +4 point(s) just awarded to @ammaratef45[/z](https://www.0crat.com/u/ammaratef45)

<!-- https://www.0crat.com/footprint/CJT4R32Q3/c7825bc3-80f5-4ab9-b259-485132b50235, version: 0.47.6, hash: ${buildNumber} -->

0crat commented:
Order was finished, quality is "good": +20 point(s) just awarded to @victornoel[/z](https://www.0crat.com/u/victornoel)

<!-- https://www.0crat.com/footprint/CJT4R32Q3/b6de9f6e-cab8-4eeb-b09f-8cae1bcf511f, version: 0.47.6, hash: ${buildNumber} -->
